src/control/server/storage/config.go

```diff
@@ -572,12 +575,120 @@ func parsePCIBusRange(numRange string, bitSize int) (uint8, uint8, error) {
 	return uint8(begin), uint8(end), nil
 }
 
+// AccelOptionBits is a type alias representing optional capabilities as a bit-set.
+type AccelOptionBits uint16
+
+// toStrings returns a slice of option names that have been set.
+func (obs AccelOptionBits) toStrings() []string {
+	opts := common.NewStringSet()
+	for str, flag := range accelOptStr2Flag {
+		if obs&flag == flag {
+			opts.Add(str)
+		}
+	}
+
+	return opts.ToSlice()
+}
+
+// fromStrings generates bit-set referenced by the function receiver from the option names provided.
+func (obs *AccelOptionBits) fromStrings(opts ...string) error {
+	if obs == nil {
+		obs = new(AccelOptionBits)
+	}
```

Comment:
This won't work the way you think it will. The reason is because you are working on a copy of the pointer, not the pointer itself. So `obs` would be initialized to a new value, but that value would only be in scope for this method. You can see how this works in the playground: https://go.dev/play/p/37M9kZO8ZGv ... The opts are set in the method, but trying to work with the variable afterward results in a segfault because it's still nil in the outer scope.

The pattern of checking for a nil receiver makes sense for types that could legitimately be nil, but is noise otherwise, IMO. I don't think this type could ever be nil except in the case of a programming error, given the way it's used in the code. At any rate, this should be either removed completely in order to avoid any future confusion or at least changed to return an error.

Reply:
done (returns error)